x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/processes/process_row_charts.tsx

```diff
@@ -67,16 +73,38 @@ const EmptyChartPlaceholder = ({ metricName }: { metricName: string }) => (
 
 export const ProcessRowCharts = ({ command, hasCpuData, hasMemoryData }: Props) => {
   const { request$ } = useRequestObservable();
-  const { loading, error, response } = useProcessListRowChart(command, request$);
+  const { hostTerm, indexPattern, to } = useProcessListContext();
+  const { isActiveTab } = useTabSwitcherContext();
+
+  const { data, status, error } = useFetcher(
+    async (callApi) => {
+      const response = await callApi('/api/metrics/process_list/chart', {
+        method: 'POST',
+        body: JSON.stringify({
+          hostTerm,
+          indexPattern,
+          to,
+          command,
+        }),
+      });
+
+      return decodeOrThrow(ProcessListAPIChartResponseRT)(response);
+    },
+    [command, hostTerm, indexPattern, to],
+    {
+      requestObservable$: request$,
+      autoFetch: isActiveTab('processes'),
+    }
+  );
 
-  const isLoading = loading || !response;
+  const isLoading = isPending(status) || !data;
```

Comment:
Shouldn't we show an error if it is not pending and there's no data?

Reply:
agree, but I didn't want to change much more than I already have.